public/firebase.js

```diff
@@ -0,0 +1,218 @@
+    }).catch(function(error) {
+    });
+    
+  } 
+
+//add post user
+function addUserPost(){
+  let user = firebase.auth().currentUser;
+  let post = document.getElementById("addNewPost").value;
+  db.collection("post").add({
+    post: post,
+    uid: user.uid,
+    user: user.displayName,
+    imagen: user.photoURL
+})
+.then(function(docRef) {
+  console.log("Document written with ID: ", docRef.id);  
+  document.getElementById("addNewPost").value = '';
+    showPostUser();
+    
+})
+.catch(function(error) {
+    console.error("Error adding document: ", error);
+});
+}
+
+//Show post user
+function showPostUser(){
+  const postContainer = document.getElementById('allNewPost');
+        postContainer.innerHTML='';
+        db.collection("post").onSnapshot((querySnapshot) => {
+        querySnapshot.forEach((doc) => {;
+        //console.log(`${doc.id} => ${doc.data().post}`);
+        let postElement = document.createElement('div');
+        let postNew = `
+        <div class="containerPostPrint">
+        <div class="containerUser">
+          <img class="imgUser" src="img/user.svg"></img> <p class="nameUser">${doc.data().user}</p>
+          </div>
+          <div class="btnIcon">
+          <button type="submit" class="editBtn" ${doc.id}, ${doc.data().post} ><img src="img/edit.svg" /></button> 
+          <button type="submit" class="deleteBtn" data-id="${doc.id}"><img data-id="${doc.id}" src="img/delete.svg" /></button> 
+          </div>
+          <p class="PostPrint">${doc.data().post}</p>
+          <textarea class="answer" id="answer"> </textarea>
+          <br>
+          <div class="sectionLikes">
+          <p class="nroLikes">0<img class="imgLikes" src="img/like.svg" /></p>
+          </div>
+          <button class="btnAnswer" data-id="${doc.id}">Responder</button>
+        </div>
+        `
+        postElement.innerHTML = postNew;
+        postContainer.appendChild(postElement);    
+
+  const buttonsDeletePost = document.querySelectorAll('.deleteBtn')
+        buttonsDeletePost.forEach(btn=>btn.addEventListener('click', deletePost))
+
+  const buttonAnswer = document.querySelectorAll('.btnAnswer')
+        buttonAnswer.forEach(btnA=>btnA.addEventListener('click', answer))
+
+      })
+  })
+ }
+
+ // reply post
+
+ function answer(e){
+   let user = firebase.auth().currentUser;
+   let id = e.target.getAttribute('data-id');
+   let postRef = db.collection('post').doc(id);
+   let anwerUser = document.getElementById("answer").value;
+   postRef.add({
+     answerUser: anwerUser,
+     uid:user.uid,
+     user: user.displayName
+   })
+  .then (function(){
+    console.log("reply post");
+  })
+  .catch(function(error){
+    console.log("error reply post", error);
+  })
+ }
+// Delete posts
+  function deletePost (e) {
+   let id = e.target.getAttribute('data-id')
+   console.log(e.target)
+   console.log(id)
+   if (!confirm("¿Seguro que quieres eliminar esta publicación?")) return
+
+  db.collection("post")
```

Comment:
cuando nombren sus colecciones, sugiero sea en plural 
```suggestion
  db.collection("posts")
```